### PR TITLE
Use IMDSv2

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Description

Also see the [`goformation` change](https://github.com/weaveworks/goformation/commit/18df1c671d26faf440a55170f0cf32d16aee88f9)

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

File: pkg/nodebootstrap/generate.go [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package nodebootstrap
```
Comment (martina-if) [inline pkg/nodebootstrap/generate.go:1]: What is this for?

Comment (michaelbeaumont) [reply pkg/nodebootstrap/generate.go:1]: ah woops i ended up making a separate PR for this

**Comment by michaelbeaumont** [comment]
Yeah I'm waiting on the repo to be under `weaveworks`

**Comment by michaelbeaumont** [comment]
@cPu1 are these changes enough to be "using imdsv2, not v1"?

**Comment by michaelbeaumont** [comment]
Should [v2 be required in the launch template](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html#configuring-IMDS-new-instances)?

**Comment by cPu1** [comment]
> Should [v2 be required in the launch template](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html#configuring-IMDS-new-instances)?

Not yet I think. We should wait for some time before requiring IMDSv2 on new instances.

**Pull request merged**